h/models/group_scope.py

```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+import sqlalchemy as sa
+
+from h.db import Base
+
+
+class GroupScope(Base):
+    __tablename__ = 'groupscope'
+    id = sa.Column(sa.Integer, autoincrement=True, primary_key=True)
+
+    #: A fully qualified domain name, e.g. example.com, www.nytimes.com or
+    #: web.hypothes.is.
+    hostname = sa.Column('hostname', sa.UnicodeText, nullable=False, unique=True)
```

Comment:
Given the `unique` constraint, does that mean that when creating a `GroupScope`, e.g.

```
my_scope = models.GroupScope(hostname=u'biopub.org')
```

You'd need to wrap in a `try/except`? Or, what's the pattern here (new to SQLAlchemy)?

Reply:
I don't think you'd get an exception until you added the duplicate GroupScope to the DB and tried to flush the DB. So you could try to create the GroupScope and then have an `except` and find the existing scope if it turns out it already exists, but you'd probably need to `flush()` inside the `try`.

What might work out nicer is to look for the existing GroupScope in the DB first and then if you don't find it, create the new one. Avoids any flushing.

Although now I'm worrying about concurrent transactions.

Reply:
This is how I have it currently (in code intended for a future pr):

```python
try:            
    groupscope = self.session.query(GroupScope).filter_by(hostname=hostname).one()
except sa.orm.exc.NoResultFound:
    groupscope = GroupScope(hostname=hostname)
return groupscope
```